### PR TITLE
Fix errno errors

### PR DESCRIPTION
**Pull request by sigvartmh**
Fixed the "errno" and "header" comments. Also added some more error checking for when to delete the firmware on write.

### CODE REVIEW

**Comment by sigvartmh** [comment]
pling plong @hakonfam

File: subsys/dfu/src/dfu_target_modem.c [hunk 1]
```diff
@@ -168,16 +169,27 @@ int dfu_target_modem_write(const void *const buf, size_t len)
 		return 0;
 	}
 
-	if (sent == ENOEXEC) {
+	if (errno == ENOEXEC) {
```
Comment (hakonfam) [inline subsys/dfu/src/dfu_target_modem.c:172]: Does this compile? `errno` here and `err` declared?

File: subsys/dfu/src/dfu_target_modem.c [hunk 2]
```diff
@@ -65,13 +65,12 @@ static int delete_banked_modem_fw(void)
 
 	while (true) {
 		err = getsockopt(fd, SOL_DFU, SO_DFU_OFFSET, &offset, &len);
-		if (err != 0) {
-			if (err == ENOEXEC) {
+		if (err < 0) {
+			if (errno == ENOEXEC) {
```
Comment (hakonfam) [inline subsys/dfu/src/dfu_target_modem.c:69]: Does this compile? `errno` here and `err` declared?

**Review by hakonfam — CHANGES_REQUESTED** [review]

File: subsys/dfu/src/dfu_target_modem.c (hunk 2 above)
Comment (sigvartmh) [reply subsys/dfu/src/dfu_target_modem.c:69]: `errno` is unfortunately a global 🗡👎 which gets set for each call to the modem.

Comment (sigvartmh) [reply subsys/dfu/src/dfu_target_modem.c:69]: so yes it compiles. We can't change the design decision of bsdlib/modem

**Pull request merged**